aws/src/test/java/org/apache/iceberg/aws/s3/MinioUtil.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws.s3;
+
+import java.net.URI;
+import org.testcontainers.containers.MinIOContainer;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3ClientBuilder;
+
+public class MinioUtil {
+
+  public static MinIOContainer createContainer() {
+    return createContainer(null);
+  }
+
+  public static MinIOContainer createContainer(AwsCredentials credentials) {
+    var container = new MinIOContainer(DockerImageName.parse("minio/minio:latest"));
+
+    // this enables virtual-host-style requests. see
+    // https://github.com/minio/minio/tree/master/docs/config#domain
+    container.withEnv("MINIO_DOMAIN", "localhost");
+
+    if (credentials != null) {
+      container.withUserName(credentials.accessKeyId());
+      container.withPassword(credentials.secretAccessKey());
+    }
+
+    return container;
+  }
+
+  public static S3Client createS3Client(MinIOContainer container) {
+    URI uri = URI.create("http://127.0.0.1:" + container.getFirstMappedPort());
```

Comment:
shouldn't this be using `container.getS3URL()` here rather than manually constructing the URI?

Reply:
I tried using "container.getS3URL" but it does not work for me (I am using MacOS)

Code:
```
URI uri = URI.create(container.getS3URL());
```

yields

```
Received an UnknownHostException when attempting to interact with a service. See cause for the exact endpoint that is failing to resolve. If this is happening on an endpoint that previously worked, there may be a network connectivity issue or your DNS cache could be storing endpoints for too long.
software.amazon.awssdk.core.exception.SdkClientException: Received an UnknownHostException when attempting to interact with a service. See cause for the exact endpoint that is failing to resolve. If this is happening on an endpoint that previously worked, there may be a network connectivity issue or your DNS cache could be storing endpoints for too long.
	at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:111)
```

Reply:
can you push that change please to see whether it works on CI?

Reply:
> can you push that change please to see whether it works on CI?

pushed.

Reply:
GitHub Actions CI passed.  But my laptop (MacOS) is unhappy with 
```
URI.create(container.getS3URL())
```

Exception on MacOS:
```
Received an UnknownHostException when attempting to interact with a service. See cause for the exact endpoint that is failing to resolve. If this is happening on an endpoint that previously worked, there may be a network connectivity issue or your DNS cache could be storing endpoints for too long.
software.amazon.awssdk.core.exception.SdkClientException
```